### ISSUE TITLE
How to display only alphanumeric key without any meta keys like space?

### ISSUE THREAD

**Issue by zenny** [wavexx/screenkey#23]
Hi,

How to display only alphanumeric key in screenkey without any meta keys like space and backspace as if it is normally typed in plain English?

Happy holidays!

Cheers,
/z

**Comment by wavexx** [comment]
I'm not sure I follow. Can you provide an example or mock-up image?

**Comment by zenny** [comment]
What I meant was, instead of showing http://picpaste.com/PfB1YwyF.png, 
can it be displayed like: http://picpaste.com/XJA880vj.png ?

**Comment by wavexx** [comment]
On 29/12/15 22:27, zenny wrote:

> What I meant was, instead of showing http://picpaste.com/PfB1YwyF.png,
> can it be displayed like: http://picpaste.com/XJA880vj.png ?

Incidentally, that's how the original screenkey used to work.

What should happen with enter in this case?
Just output a space?

Out of curiosity, what's your use-case scenario for this?

If I'm teaching how to use a program, I most definitely want all meta
stuff to be visible. It's very rare that I need just to show a sentence.
Is that what you'd like to do?

**Comment by zenny** [comment]
@wavexx You have precisely stated the usage scenario. For teaching, screenkey is already great with key-mon except for aged learners, as far as my experience tells.

It would be best if one can toggle the meta-keys just not to confuse aged adult learners. The extra keys are found to be confusing as they are not used to.

It is my wishful thinking that it would be better if there is a toggle in the preferences so that one can opt between. What do you think?

**Comment by wavexx** [comment]
On 30/12/15 17:44, zenny wrote:

> It would be best if one can toggle the meta-keys just not to confuse
> aged adult learners. The extra keys are found to be confusing as they
> are not used to.
> 
> It is my wishful thinking that it would be better if there is a toggle
> in the preferences so that one can opt between. What do you think?

I have no objections.

What ENTER should do in this case though?
Just show a space?

Maybe I could actually display 2 lines...

**Comment by zenny** [comment]
> I have no objections.

:D

> What ENTER should do in this case though?
> Just show a space?
> 
> Maybe I could actually display 2 lines...

You replied yourself. Genius!

**Comment by wavexx** [comment]
Humm, I need to rethink how font-size works. I usually shrink the text to fit the label, but with multiline input it's not really good if you have more than 2 lines.

If I enlarge the window, size might _also_ be an issue.
Could you give it a shot?

Use `./screenkey --multiline` for some testing.

**Comment by wavexx** [comment]
Try with `--multiline --no-whitespace`. Font sizing is still bad, but I'm interested about the behavior.

**Comment by zenny** [comment]
§1. --multiline --no-whitespace removes the whitespace between the words, but it does not remove Backspace, Del and Ctrl+alphanumeric keys among others while typing, but not when editing the already typed letters.

§2. On the other hand multiple lines automagically reduces the fontsize to 50% against each line exponentially, fyi as seen at http://picpaste.com/CSuwYnsB.png

**Comment by wavexx** [comment]
On 31/12/15 16:37, zenny wrote:

> §1. --multiline --no-whitespace removes the whitespace between the
> words, but it does not remove Backspace, Del and Ctrl+alphanumeric keys
> among others while typing, but not when editing the already typed letters.

Backspace is also invisible when un-ambiguous.
--bak-mode=full will also allow you to backspace through newlines
without showing the backspace symbol.

Isn't this enough?

Hiding all meta keys might not always do the right thing.
For instance, if you press Ctrl+Enter, it will completely be ignored.

I wonder if just toggling recording (with ctrl+ctrl) would be better
instead. I do this a lot in my sessions to avoid cluttering the screen.

> §2. On the other hand multiple lines automagically reduces the fontsize
> to 50% against each line exponentially, fyi as seen at
> http://picpaste.com/CSuwYnsB.png

Yes, I still need to fix this.
I wanted to see how it well it works first.

**Comment by Zerokami** [comment]
For future reference to myself and other and for Google searchers.
The solution for removing space keys is adding `--no-whitespace` and to add mutliline add `--multiline`
For example `screenkey --no-whitespace --multiline`
